src/core/gltf2importer/animationparser.cpp

```diff
@@ -173,6 +173,19 @@ QString channelPathToProperty(const QString &path)
     return {};
 }
 
+quint8 expectedComponentsCountForChannel(const QString &channelName)
+{
+    if (channelName == QStringLiteral("translation"))
+        return 3;
+    if (channelName == QStringLiteral("rotation"))
+        return 4;
+    if (channelName == QStringLiteral("scale"))
+        return 3;
+    if (channelName == QStringLiteral("weights"))
+        return 1;
+    return 0;
```

Comment:
might be good, if the result is 0, to print a warning that the channel name is invalid

Reply:
done